src/result-async.ts

```diff
@@ -68,6 +68,19 @@ export class ResultAsync<T, E> implements PromiseLike<Result<T, E>> {
     ) as CombineResultsWithAllErrorsArrayAsync<T>
   }
 
+  tap(f: (t: T) => void | Promise<void>): ResultAsync<T, E> {
+    return new ResultAsync(
+      this._promise.then(async (res: Result<T, E>) => {
+        if (res.isErr()) {
+          return new Err<T, E>(res.error)
+        }
+
+        await f(res.value)
```

Comment:
What if the promise fails? Shouldn't this be wrapped in a try / catch? `ResultAsync.fromPromise` could also be used here, which requires a handler function in the event the promise fails.

Reply:
Good question, I'm honestly not sure. Maybe that's a good indicator that I should add a test that covers that(?)

I pretty much copied the `map` implementation. The only difference is that instead of returning the result of the passed function, I execute the function and then return the previous value.

If I add a try-catch to the tap function, shouldn't I also add one to the `map` and `mapErr` functions? 🤔